sui/src/sui_genesis.rs

```diff
@@ -0,0 +1,213 @@
+// Copyright (c) 2022, Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use anyhow::anyhow;
+use std::{
+    collections::{BTreeMap, BTreeSet},
+    path::PathBuf,
+    sync::Arc,
+};
+use tokio::sync::mpsc::channel;
+
+use move_binary_format::CompiledModule;
+use move_package::BuildConfig;
+use sui_adapter::{adapter::generate_package_id, genesis};
+use sui_config::{sui_config_dir, GenesisConfig, NetworkConfig};
+use sui_core::{
+    authority::{AuthorityState, AuthorityStore, ReplicaStore},
+    authority_server::AuthorityServer,
+    full_node::FullNodeState,
+};
+use sui_types::{
+    base_types::{ObjectID, SequenceNumber, SuiAddress, TxContext},
+    committee::Committee,
+    error::SuiResult,
+    object::Object,
+};
+use tracing::info;
+
+use crate::keystore::{Keystore, SuiKeystore};
+
+pub struct GenesisState {
```

Comment:
Given https://github.com/MystenLabs/sui/pull/1972 how difficult do you think it would be to rebase on top of and use that genesis mechanism?

Reply:
Looking at the code, it shouldn't be too bad, but I'm hesitant to rebase until it merges so I don't have to keep rebasing.
In the meantime, the Genesis portion of this PR is not really relevant to the core functionality though so I can it can still be reviewed independently of that